tests/phpunit/TestFramework/PhpUnit/Adapter/PhpUnitAdapterTest.php

```diff
@@ -210,6 +199,60 @@ public function test_it_provides_initial_test_run_command_line_when_coverage_rep
         );
     }
 
+    #[Group('integration')]
+    public function test_it_provides_initial_test_run_command_line_with_fast_path_when_coverage_report_is_requested(): void
+    {
+        $this->adapter = $this->getPHPUnitAdapter('12.5');
+
+        $this->cliArgumentsBuilder
+            ->expects($this->once())
+            ->method('buildForInitialTestsRun')
+            ->with('', '--group=default --exclude-source-from-xml-coverage --coverage-xml=/tmp/coverage-xml --log-junit=/tmp/infection/junit.xml')
+            ->willReturn([
+                '--group=default', '--exclude-source-from-xml-coverage --coverage-xml=/tmp/coverage-xml', '--log-junit=/tmp/infection/junit.xml',
+            ])
+        ;
+
+        $this->commandLineBuilder
+            ->expects($this->once())
+            ->method('build')
+            ->with('/path/to/phpunit', ['-d', 'memory_limit=-1'], [
+                '--group=default', '--exclude-source-from-xml-coverage --coverage-xml=/tmp/coverage-xml', '--log-junit=/tmp/infection/junit.xml',
+            ])
+            ->willReturn([
+                '/path/to/phpunit',
+                '--group=default',
+                '--exclude-source-from-xml-coverage',
+                '--coverage-xml=/tmp/coverage-xml',
+                '--log-junit=/tmp/infection/junit.xml',
+            ])
+        ;
+
+        $this->pcovDirectoryProvider
+            ->expects($this->once())
+            ->method('shallProvide')
+            ->willReturn(false)
+        ;
+
+        $this->pcovDirectoryProvider
+            ->expects($this->never())
+            ->method('getDirectory')
+        ;
+
+        $initialTestRunCommandLine = $this->adapter->getInitialTestRunCommandLine('--group=default', ['-d', 'memory_limit=-1'], false);
+
+        $this->assertSame(
+            [
+                '/path/to/phpunit',
+                '--group=default',
+                '--exclude-source-from-xml-coverage',
+                '--coverage-xml=/tmp/coverage-xml',
+                '--log-junit=/tmp/infection/junit.xml',
+            ],
+            $initialTestRunCommandLine,
+        );
+    }
+
     #[Group('integration')]
     public function test_it_provides_initial_test_run_command_line_when_coverage_report_is_requested_and_pcov_is_in_use(): void
```

Comment:
initially I tried to use a data-provider on this test, but because of the exzessiv use of mocks we nearly need to change every line. therefore a data-provider would need a lot of parameters and this would turn this test into a unreadable mess.

therefore I decided to add a separate test to not overcomplicate things.